src/zmalloc.c

```diff
@@ -87,10 +87,26 @@ void zlibc_free(void *ptr) {
 #define dallocx(ptr, flags) je_dallocx(ptr, flags)
 #endif
 
-#define update_zmalloc_stat_alloc(__n) atomic_fetch_add_explicit(&used_memory, (__n), memory_order_relaxed)
-#define update_zmalloc_stat_free(__n) atomic_fetch_sub_explicit(&used_memory, (__n), memory_order_relaxed)
+static _Atomic int64_t used_memory = 0;
+static __thread int64_t thread_used_memory_delta = 0;
```

Comment:
`__thread` is a compiler extension. We can use standard C11 `thread_local` if we include `<threads.h>`. See https://en.cppreference.com/w/c/language/storage_duration#Notes. (`_Thread_local` is another possibility C11 without include threads.h.)